eliHenderson/assignments/reversedLinkedList/lc92.js

```diff
@@ -0,0 +1,204 @@
+/**
+ * Problem:
+ *  Given: Linked List such as 1 -> 2 -> 3-> 4 -> 5
+ *  reverse as sub section of the linked list
+ *                  1 -> 2 -> 3-> 4 -> 5
+ *                      p         q
+ *
+ *
+ *
+ * edge:
+ *  this method works on sorted linked lists
+ *
+ *
+ *
+ * overview:
+ *  check if p and q are equal
+ *  if so just return the head
+ *
+ *  curr = head
+ *  prev = null
+ *
+ *
+ *  1 -> 2 -> 3-> 4 -> 5
+ *      p         q
+ *  c
+ *
+ *  the current value has to start at the sub-arr
+ *
+ *  counter = 0
+ *
+ * while curr !== null and counter < p - 1
+ *
+ *          prev  = curr
+ *          curr = curr.next
+ *          counter ++
+ *
+ *  1 <- 2 <- 3<- 4  5
+ *      p         q
+ *                  curr
+ *              prev
+ *
+ *   1  4-> 3-> 2  5
+ *
+ * counter = 0
+ *
+ * lof = prev
+ * los = curr
+ *
+ *
+ *
+ * while ( curr !=null && counter < q - p + 1 )
+ *          let next = curr.next
+ *          curr.next = prev
+ *          prev = current
+ *          curr = next
+ *          counter ++
+ *
+ *
+ *  1  4-> 3-> 2   5
+ *      p         q
+ *               curr
+ *    prev
+ *  lof
+ *            los
+ *  1 -> 4-> 3-> 2   5
+ *      p         q
+ *               curr
+ *    prev
+ *  lof
+ *            los
+ *
+ *
+ *
+ * lof.next = prev
+ * los.next = curr
+ *
+ *
+ * return head
```

Comment:
This is an opinionated topic, but, some would say that the high level overview should be more of a short story as to how you will achieve end goal. The visualization is an amazing supplement but I do not think can take the place of this psuedo code